src/compiler/parser.rs

```diff
@@ -512,8 +525,12 @@ impl ParserType {
     }
 
     fn create_node_array(&self, elements: Vec<Node>) -> NodeArray {
-        self.factory
-            .create_node_array(elements.into_iter().map(Rc::new).collect::<Vec<Rc<Node>>>())
+        self.factory.create_node_array(
+            elements
+                .into_iter()
+                .map(Node::wrap)
```

Comment:
Here's somewhere where I needed to convert `Node` -> `Rc<Node>` - I couldn't generate an `impl From<Node> for Rc<Node>` because there's already a (conflicting) blanket implementation for that, so I hand-rolled a `Node::wrap()` method to do the equivalent thing (ie encapsulate the setting of the weak reference to the node wrapper at the time of its creation)